thorlcr/master/thgraphmanager.cpp

```diff
@@ -1114,6 +1114,7 @@ bool CJobManager::executeGraph(IConstWorkUnit &workunit, const char *graphName,
             wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTgraph, graphScope, StWhenStarted, NULL, startTs, 1, 0, StatsMergeAppend);
             //Could use addTimeStamp(wu, SSTgraph, graphName, StWhenStarted, wfid) if start time could be this point
             wu->setState(WUStateRunning);
+            wu->setEngineSession(myProcessSession());
```

Comment:
Should this be cleared when the graph completes?  Otherwise I think...
* thor instance executes a graph for wu1
* eclagent starts some spray operations for wu1
* thor is terminated for some reason running wu2
The wait for workunit may spot that the thorinstance is dead and kill the current job.

Reply:
Yes, you're right. I will clear the session id.